daemon/containerd/service.go

```diff
@@ -137,6 +140,47 @@ func (i *ImageService) GetLayerFolders(img *image.Image, rwLayer layer.RWLayer)
 }
 
 // GetContainerLayerSize returns the real size & virtual size of the container.
-func (i *ImageService) GetContainerLayerSize(containerID string) (int64, int64) {
-	panic("not implemented")
+func (i *ImageService) GetContainerLayerSize(ctx context.Context, containerID string) (int64, int64, error) {
+	snapshotter := i.client.SnapshotService(containerd.DefaultSnapshotter)
+	sizeCache := make(map[digest.Digest]int64)
+	snapshotSizeFn := func(d digest.Digest) (int64, error) {
+		if s, ok := sizeCache[d]; ok {
+			return s, nil
+		}
+		usage, err := snapshotter.Usage(ctx, d.String())
+		if err != nil {
+			return 0, err
+		}
+		sizeCache[d] = usage.Size
+		return usage.Size, nil
+	}
+
+	c, err := i.client.ContainerService().Get(ctx, containerID)
+	if err != nil {
+		if cerrdefs.IsNotFound(err) {
+			return 0, 0, nil
+		}
+		return 0, 0, err
+	}
+	image, err := i.client.GetImage(ctx, c.Image)
```

Comment:
This runs into two issues:
* We can't rely on the containerd container object as we need to get the size for stopped containers.
* The same issue discussed in https://github.com/moby/moby/pull/44958#discussion_r1101404155 applies re: the resolution of the user's input image spec vs. the actual content-addressable blob in the containerd store; we need to solve that for both these PRs.